test/test_cli.rb

```diff
@@ -17,11 +17,18 @@ def test_empty_args
 
   def test_invalid_option
     err = assert_raise(OptionParser::InvalidOption) do
-      RipperTags.process_args(%[--moo])
+      RipperTags.process_args(%w[--moo])
     end
     assert_equal "invalid option: --moo", err.message
   end
 
+  def test_invalid_option_with_ignored
+    err = assert_raise(OptionParser::InvalidOption) do
```

Comment:
I don't think this test is sufficient.

First of all, we want a test that demonstrates that with `--ignore-unsupported-options`, no InvalidOption exception will be raised. For that, supply an extra argument that represents a list of files, or the `-R` flag.

Second, we want to check what happens with ignored unsupported options. We want to make sure that they are discarded instead of being treated as the list of filesystem paths to process. Therefore, `--moo lib --ignore-unsupported-options --languages=foo src` should result in the list of files: `["lib", "src"]`.